### PR TITLE
test: Add open api validation in acceptance tests

### PR DESCRIPTION
**Pull request by oakbani**
## Summary
This adds open API request and response validation in acceptance tests. 

## Testplan
Modified acceptance tests which are covering most of the scenarios and used openapi spec to validate that. All must pass.

## Issues
OASIS-6781

### CODE REVIEW

File: api/openapi-spec/openapi.yaml [hunk 1]
```diff
@@ -252,17 +250,14 @@ components:
           enum:
             - feature
             - experiment
+            - ''
         enabled:
           type: boolean
         variables:
           type: object
-          additionalProperties:
```
Comment (msohailhussain) [inline api/openapi-spec/openapi.yaml:259]: we may add Variable Component?

File: api/openapi-spec/openapi.yaml [hunk 2]
```diff
@@ -347,12 +340,12 @@ components:
...
-  securitySchemes:
-    SdkKeyAuth:
-      in: header
-      name: X-Optimizely-SDK-Key
-      type: apiKey
-    TokenAuth:
-      type: http
-      scheme: bearer
-      bearerFormat: JWT
+# securitySchemes:
```
Comment (msohailhussain) [inline api/openapi-spec/openapi.yaml:343]: Python lib not supporting securitySchemes, that's why commented out.

File: tests/acceptance/test_acceptance/conftest.py [hunk 3]
```diff
@@ -77,7 +77,7 @@ def pytest_addoption(parser):
     Adding CLI option to specify host URL to run tests on.
     :param parser: parser
     """
-    parser.addoption("--host", action="store", default="http//:localhost:8080",
+    parser.addoption("--host", action="store", default="http://localhost:8080",
```
Comment (msohailhussain) [inline tests/acceptance/test_acceptance/conftest.py:80]: :D

File: tests/acceptance/test_acceptance/test_activate.py [hunk 4]
```diff
@@ -221,16 +220,16 @@ def test_activate__feature(session_obj, feature_key, expected_response,
 ]"""
 
 
-@pytest.mark.parametrize("decision_type, expected_response, expected_status_code", [
-    ("experiment", expected_activate_type_exper, 200),
-    ("feature", expected_activate_type_feat, 200),
+@pytest.mark.parametrize("decision_type, expected_response, expected_status_code, bypass_validation", [
+    ("experiment", expected_activate_type_exper, 200, False),
```
Comment (msohailhussain) [inline tests/acceptance/test_acceptance/test_activate.py:224]: There are some request which won't pass validation but we need to check response. That's why added one more arg, to bypass validation for request.

File: api/openapi-spec/openapi.yaml (hunk 2 above)
Comment (mikecdavis) [reply api/openapi-spec/openapi.yaml:343]: A bit unfortunate to remove this.

Comment (aliabbasrizvi) [reply api/openapi-spec/openapi.yaml:343]: Can you add a comment above this about why it is still around?

File: tests/acceptance/test_acceptance/test_activate.py [hunk 5]
```diff
@@ -6,6 +6,7 @@
 from tests.acceptance.helpers import ENDPOINT_ACTIVATE
 from tests.acceptance.helpers import ENDPOINT_CONFIG
 from tests.acceptance.helpers import sort_response
+from tests.acceptance.helpers import create_and_validate_request_and_response
```
Comment (aliabbasrizvi) [inline tests/acceptance/test_acceptance/test_activate.py:9]: nit. alphabetize.

File: tests/acceptance/test_acceptance/test_config.py [hunk 6]
```diff
@@ -4,6 +4,9 @@
 import requests
 
 from tests.acceptance.helpers import ENDPOINT_CONFIG
+from tests.acceptance.helpers import create_and_validate_request
+from tests.acceptance.helpers import create_and_validate_response
+from tests.acceptance.helpers import create_and_validate_request_and_response
```
Comment (aliabbasrizvi) [inline tests/acceptance/test_acceptance/test_config.py:9]: nit. alphabetize.

File: tests/acceptance/test_acceptance/test_overrides.py [hunk 7]
```diff
@@ -7,6 +7,10 @@
 from tests.acceptance.helpers import ENDPOINT_OVERRIDE
 from tests.acceptance.helpers import activate_experiment
 from tests.acceptance.helpers import override_variation
+from tests.acceptance.helpers import create_and_validate_request
+from tests.acceptance.helpers import create_and_validate_response
+from tests.acceptance.helpers import create_and_validate_request_and_response
```
Comment (aliabbasrizvi) [inline tests/acceptance/test_acceptance/test_overrides.py:12]: nit. alphabetize imports.

File: api/openapi-spec/openapi.yaml (hunk 2 above)
Comment (msohailhussain) [reply api/openapi-spec/openapi.yaml]: It's was mistake in our code, didn't realize. fixed the issue.

**Pull request merged**